### PR TITLE
Update the notification email for completed accessioning

### LINKED ISSUE #1220: Send email when accessioning complete or results in errors
Currently users get an email when their Job has completed preaccessioning. Now that we know when an object finished accessioning (see #1201) we can let users know when their job has cleared accessioning instead. This will prevent situations where a user sees a completion email when the object actually errored durig accessioning.

Since we also know what DRUIDs were accessioned (see #1219) we can also include in the body of the email which DRUIDs have completed, with links to Argo.

```
Dear {user},

Your Preassembly job {link-to-preassembly-job} has successfully accessioned the following SDR objects:

- https://argo.stanford.edu/view/druid:123
- https://argo.stanford.edu/view/druid:456
- https://argo.stanford.edu/view/druid:789

The following SDR items failed accessioning:

- https://argo.stanford.edu/view/druid:000
```

### PR DESCRIPTION
**Pull request by aaron-collier**
# Why was this change made? 🤔

Closes #1220 

Note: this has been rebased on main after https://github.com/sul-dlss/pre-assembly/pull/1281/

# How was this change tested? 🤨

Unit

# Does your change introduce accessibility violations? 🩺

no

### CODE REVIEW

**Comment by ndushay** [comment]
Is this PR waiting for anything?

File: config/settings.yml
Comment (ndushay) [inline config/settings.yml:15]: is this actually used?

**Review by ndushay — APPROVED** [review]
approve modulo one question.

**Comment by ndushay** [comment]
how does this relate to #1272

File: app/mailers/job_mailer.rb
Comment (jmartin-sul) [inline app/mailers/job_mailer.rb:15]: i think you could do this in query-land without loading all the `accessions` and using a ruby `select`, by doing something like `@job_run.accessions.where(state: 'completed')` (and i think [ActiveRecord's `enum` actually provides some scope magic](https://edgeapi.rubyonrails.org/classes/ActiveRecord/Enum.html) that'd let you do e.g. `@job_run.accessions.completed`).

i'm not sure how many `Accessions` we expect to hang off of a `JobRun`.  probably few enough that this doesn't really matter from a performance PoV, and what's here is still readable, so def wouldn't request changes over this, just mentioning for consideration.

File: app/mailers/job_mailer.rb
Comment (jmartin-sul) [inline app/mailers/job_mailer.rb:19]: same filtering/query observation here

File: app/mailers/job_mailer.rb
Comment (jmartin-sul) [reply app/mailers/job_mailer.rb:15]: actually, looking at the model/schema and the rest of the code, i'm remembering that an `Accession` is one druid, not the whole batch being accessioned by the job.  so actually it might be noticeably better to filter on the SQL query side since there could be up to a few hundred or even a few thousand associated druids?  still wouldn't block the PR over this, though.  retrieving e.g. 5k rows on occasion in an async job is unlikely to crash things.

File: config/settings.yml
Comment (jmartin-sul) [reply config/settings.yml:15]: i see it referenced in the email erb, e.g. https://github.com/sul-dlss/pre-assembly/pull/1284/files#diff-6e5dcc75b9e9652d1202ebedadc436e5e46d2ec6968aa8752d28cf67e82e9af2R12

File: spec/mailers/job_mailer_spec.rb
Comment (jmartin-sul) [inline spec/mailers/job_mailer_spec.rb:42]: iirc, i think you can also set `job_type` to the string value (`'preassembly'`), and activerecord magic will do the translation for you, which might make this slightly more readable.

File: spec/mailers/job_mailer_spec.rb
Comment (jmartin-sul) [inline spec/mailers/job_mailer_spec.rb:4]: neat, didn't know about this!

**Review by jmartin-sul — APPROVED** [review]
a couple nitpicks for your consideration, but otherwise LGTM

File: config/settings.yml
Comment (ndushay) [reply config/settings.yml:15]: ah, ok.

**Review by jmartin-sul — APPROVED** [review]
thanks for the revisit, approved again 🙂

**Pull request merged**
